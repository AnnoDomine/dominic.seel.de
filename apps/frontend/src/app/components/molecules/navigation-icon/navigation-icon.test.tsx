import { render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it } from "vitest";

describe("NavigationIcon", () => {
    // Use a helper function to mock useUser for clarity
    const mockUseUser = (isAuthenticated: boolean) => {
        vi.doMock("../../../../redux/hooks/useUser.hooks", () => ({
            __esModule: true,
            default: vi.fn(() => ({
                isAuthenticated,
            })),
        }));
    };

    beforeEach(() => {
        vi.clearAllMocks();
        vi.resetModules();
        mockUseUser(false);
        vi.mock("../../../../redux/queries/auth", () => ({
            useLoginMutation: vi.fn(() => [vi.fn(), { isLoading: false }]),
        }));
        vi.mock("../../../../redux/queries/user", () => ({
            useGetUserQuery: vi.fn(() => ({ data: null, refetch: vi.fn() })),
            useLogoutMutation: vi.fn(() => [vi.fn(), { isLoading: false }]),
        }));
    });
    afterEach(() => {
        vi.clearAllMocks();
        vi.restoreAllMocks();
    });

    it("renders the correct icon for 'home", async () => {
        const { default: NavigationIcon } = await import("./navigation-icon");
        render(<NavigationIcon type="home" />);
        const navigationIcon = screen.getByTestId("HomeTwoToneIcon");

        expect(navigationIcon).toMatchSnapshot();
    });

    it("renders the correct icon for 'projects'", async () => {
        const { default: NavigationIcon } = await import("./navigation-icon");
        render(<NavigationIcon type="projects" />);
        const navigationIcon = screen.getByTestId("AccountTreeTwoToneIcon");

        expect(navigationIcon).toMatchSnapshot();
    });

    it("renders the correct icon for 'roadmap'", async () => {
        const { default: NavigationIcon } = await import("./navigation-icon");
        render(<NavigationIcon type="roadmap" />);
        const navigationIcon = screen.getByTestId("RouteTwoToneIcon");

        expect(navigationIcon).toMatchSnapshot();
    });

    it("renders the correct icon for 'admin-control-panel' when not logged in", async () => {
        const { default: NavigationIcon } = await import("./navigation-icon");
        render(<NavigationIcon type="acp" />);
        const navigationIcon = screen.getByTestId("HttpsTwoToneIcon");

        expect(navigationIcon).toMatchSnapshot();
    });

    it("renders the correct icon for 'admin-control-panel' when logged in", async () => {
        mockUseUser(true);
        const { default: NavigationIcon } = await import("./navigation-icon");
        render(<NavigationIcon type="acp" />);
        const navigationIcon = screen.getByTestId("LockOpenTwoToneIcon");

        expect(navigationIcon).toMatchSnapshot();
    });
});
