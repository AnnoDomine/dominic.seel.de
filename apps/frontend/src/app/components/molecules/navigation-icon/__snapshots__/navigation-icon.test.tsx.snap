// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationIcon > renders the correct icon for 'admin-control-panel' when logged in 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
  data-testid="LockOpenTwoToneIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M6 20h12V10H6zm6-7c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2"
    opacity=".3"
  />
  <path
    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h2c0-1.66 1.34-3 3-3s3 1.34 3 3v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m0 12H6V10h12zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2"
  />
</svg>
`;

exports[`NavigationIcon > renders the correct icon for 'admin-control-panel' when not logged in 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
  data-testid="HttpsTwoToneIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M6 20h12V10H6zm6-7c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2"
    opacity=".3"
  />
  <path
    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2M9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9zm9 14H6V10h12zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2"
  />
</svg>
`;

exports[`NavigationIcon > renders the correct icon for 'home 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
  data-testid="HomeTwoToneIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M12 3 2 12h3v8h6v-6h2v6h6v-8h3zm5 15h-2v-6H9v6H7v-7.81l5-4.5 5 4.5z"
  />
  <path
    d="M7 10.19V18h2v-6h6v6h2v-7.81l-5-4.5z"
    opacity=".3"
  />
</svg>
`;

exports[`NavigationIcon > renders the correct icon for 'projects' 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
  data-testid="AccountTreeTwoToneIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3zM7 9H4V5h3zm10 6h3v4h-3zm0-10h3v4h-3z"
  />
  <path
    d="M7 5v4H4V5zm13 0v4h-3V5zm0 10v4h-3v-4z"
    opacity=".3"
  />
</svg>
`;

exports[`NavigationIcon > renders the correct icon for 'roadmap' 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
  data-testid="RouteTwoToneIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <circle
    cx="6"
    cy="6"
    opacity=".3"
    r="1"
  />
  <circle
    cx="18"
    cy="18"
    opacity=".3"
    r="1"
  />
  <path
    d="M19 15.18V7c0-2.21-1.79-4-4-4s-4 1.79-4 4v10c0 1.1-.9 2-2 2s-2-.9-2-2V8.82C8.16 8.4 9 7.3 9 6c0-1.66-1.34-3-3-3S3 4.34 3 6c0 1.3.84 2.4 2 2.82V17c0 2.21 1.79 4 4 4s4-1.79 4-4V7c0-1.1.9-2 2-2s2 .9 2 2v8.18c-1.16.41-2 1.51-2 2.82 0 1.66 1.34 3 3 3s3-1.34 3-3c0-1.3-.84-2.4-2-2.82M6 7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1m12 12c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1"
  />
</svg>
`;
